language_service/src/completion.rs

```diff
@@ -272,9 +277,9 @@ impl CompletionListBuilder {
     fn push_globals(
         &mut self,
         compilation: &Compilation,
-        opens: &[(Rc<str>, Option<Rc<str>>)],
+        opens: &[(NamespaceName, Option<NamespaceAlias>)],
         insert_open_range: Option<Range>,
-        current_namespace_name: &Option<Rc<str>>,
+        current_namespace_name: &Option<Vec<Rc<str>>>,
```

Comment:
AFAICT you're not changing any functionality in completions. Wouldn't it be much more straightforward and backwards-compatible if you left these as `Rc<str>`, and initialized them to the dotted names (`namespace.name.name()`)? `get_callables` doesn't need to have awareness of these `VecIdent`s since it's not doing anything special with them.